kanidm_unix_int/src/daemon.rs

```diff
@@ -393,29 +443,23 @@ async fn main() {
             debug!("Profile -> {}", env!("KANIDM_PROFILE_NAME"));
             debug!("CPU Flags -> {}", env!("KANIDM_CPU_FLAGS"));
 
-            let cfg_path = Path::new("/etc/kanidm/config");
-            let cfg_path_str = match cfg_path.to_str() {
-                Some(cps) => cps,
-                None => {
-                    error!("Unable to turn cfg_path to str");
-                    std::process::exit(1);
-                }
-            };
+            // we can do a blind unwrap because there's a default in clap
+            let cfg_path_str = clap_args.get_one::<String>("client-config").unwrap();
+            let cfg_path: PathBuf =  PathBuf::from(cfg_path_str);
+
             if !cfg_path.exists() {
                 // there's no point trying to start up if we can't read a usable config!
                 error!(
                     "Client config missing from {} - cannot start up. Quitting.",
                     cfg_path_str
                 );
-                std::process::exit(1);
-            }
-
-            if cfg_path.exists() {
+                return
+            } else {
                 let cfg_meta = match metadata(&cfg_path) {
                     Ok(v) => v,
                     Err(e) => {
                         error!("Unable to read metadata for {} - {:?}", cfg_path_str, e);
-                        std::process::exit(1);
+                        return
```

Comment:
https://doc.rust-lang.org/std/process/struct.ExitCode.html

You can return exitcodes now!

Reply:
Yas, but since this is hidden inside the tracing worker magic... then inside tokio's wrapped event loop thing, I can't work out what to do... :sad:

Reply:
:(